src/ds5/ds5-options.h

```diff
@@ -398,4 +398,32 @@ namespace librealsense
         sensor_base* _sensor;
     };
 
+
+    class ds5_thermal_monitor;
+    class thermal_compensation : public option
+    {
+    public:
+        thermal_compensation(std::shared_ptr<ds5_thermal_monitor> monitor,
+            std::shared_ptr<option> toggle);
+
+        void set(float value) override;
+        float query() const override;
+
+        option_range get_range() const override { return option_range{ 0, 1, 1, 0 }; }
+        bool is_enabled() const override { return true; }
+
+        const char* get_description() const override;
+        const char* get_value_description(float value) const override;
+
+        void enable_recording(std::function<void(const option&)> record_action) override
+        {
+            _recording_function = record_action;
+        }
+
+    private:
+        mutable std::shared_ptr<ds5_thermal_monitor>  _thermal_monitor;
```

Comment:
Please explain the reason for the `mutable`

Reply:
The use-case is toggling ThermalLoop from external application, so that the control's value in FW is volatile and the SDK's internal state responds to this event

Reply:
I don't follow. The only reason I see is the `const` function that calls `update` -- where I don't see why it's needed. Let's talk.